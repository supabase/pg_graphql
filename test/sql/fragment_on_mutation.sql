begin;

    create table blog_post(
        id int primary key,
        title text not null
    );

    select graphql.resolve($$
        mutation {
          ...blogPosts_insert
        }

        fragment blogPosts_insert on Mutation {
          insertIntoBlogPostCollection(objects: [
            { id: 1, title: "foo" }
          ]) {
            affectedCount
            records {
              id
              title
            }
          }
        }
    $$);

    select * from blog_post;

rollback;
