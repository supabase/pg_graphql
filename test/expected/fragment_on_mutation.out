begin;
    create table blog_post(
        id int primary key,
        title text not null
    );
    select graphql.resolve($$
        mutation {
          ...blogPosts_insert
        }

        fragment blogPosts_insert on Mutation {
          insertIntoBlogPostCollection(objects: [
            { id: 1, title: "foo" }
          ]) {
            affectedCount
            records {
              id
              title
            }
          }
        }
    $$);
                                                 resolve                                                  
----------------------------------------------------------------------------------------------------------
 {"data": {"insertIntoBlogPostCollection": {"records": [{"id": 1, "title": "foo"}], "affectedCount": 1}}}
(1 row)

    select * from blog_post;
 id | title 
----+-------
  1 | foo
(1 row)

rollback;
